src/page-modules/assistant/layout.tsx

```diff
@@ -97,6 +98,7 @@ function AssistantLayout({
   const onSubmitHandler: FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
     if (!selectedFromFeature || !selectedToFeature) return;
+    setIsSearching(true);
```

Comment:
Should `isSearching()` be set when swapping also?

Reply:
I refactored this so that `isSwapping` is only true when we are actually swapping. Then, when we start the search (i.e. `router.push`) we set `isSearching` to true. Having both buttons spin at the same time was a bit confusing, but I think this solution is more clear 👍